src/app/components/DevSiteRoutingMenu.jsx

```diff
@@ -0,0 +1,211 @@
+
+const propTypes = {
+  /**
+   * The title to render within the DevSiteRoutingMenu's header.
+   */
+  title: PropTypes.string,
+  /**
+   * The array of routing shapes to be rendered as menu items within the DevSiteRoutingMenu.
+   */
+  menuItems: PropTypes.arrayOf(PropTypes.shape({
+    text: PropTypes.string.isRequired,
+    path: PropTypes.string.isRequired,
+    hasSubMenu: PropTypes.bool,
+  })),
+  /**
+   * The Object of layout-related APIs provided to the components of the Layout.
+   */
+  layoutConfig: PropTypes.shape({
+    size: PropTypes.string,
+    toggleMenu: PropTypes.func,
+    menuIsOpen: PropTypes.bool,
+    togglePin: PropTypes.bool,
+    menuIsPinned: PropTypes.bool,
+  }).isRequired,
+  /**
+   * The Object containing RoutingStack APIs provided to children of the RoutingStack. This is
+   * used to render a Back button upon presence of a `showParent` implementation.
+   */
+  routingStackDelegate: RoutingStackDelegate.propType.isRequired,
+  /**
+   * The location from the router context. This prop is provided by the `withRouter` HOC-generator.
+   */
+  location: PropTypes.shape({
+    pathname: PropTypes.string,
+  }).isRequired,
+};
+
+const defaultProps = {
+  title: '',
+  menuItems: [],
+};
+
+const devSiteRoutingMenuRootMenuKey = 'devSiteRoutingMenuRootMenuKey';
+const FILTER_DISPLAY_THRESHOLD = 10;
+
+/**
+ * This function is curried and allows you to quickly determine if the searchable contains a case-insensitive match with
+ * by.
+ * @param {string} by is the string to match by
+ * @returns {(searchable?: string) => Boolean}
+ */
+export const searchFilter = (by = '') => (searchable = '') => searchable.toLowerCase().search(by.toLowerCase()) >= 0;
+
+/**
+ * This function converts the given menuItems from the DevSiteRoutingMenu API to the NavigationSideMenu API.
+ * The path is used to uniquely identify the item within the NavigationSideMenu. The path and hasSubMenu
+ * values are set as metaData on the item so that `handleMenuChange` will have easy access to those values.
+ * @param {Array} items is the Array of menuItem objects as specified by the DevSiteRoutingMenu's proptype definition.
+ */
+export const buildSideMenuItems = (items = []) => items.map(({
+  path, text, hasSubMenu, externalLink,
+}) => ({
+  key: path,
+  text,
+  hasSubMenu: !!hasSubMenu,
+  metaData: {
+    path,
+    externalLink,
+    hasSubMenu: !!hasSubMenu,
+  },
+}));
+
+/**
+ * This function compares the given path against the paths of the given menuItems. If a match
+ * (partial or otherwise) is detected, that path is returned. If no match is detected, `undefined` is returned.
+ * @param {String} pathToMatch is the path to be matched against
+ * @param {Array} menuItems is the Array of menuItem objects as specified by the DevSiteRoutingMenu's proptype definition.
+ */
+const getSelectedChildKey = (pathToMatch = '', menuItems = []) => (menuItems.filter(({ path }) => path && matchPath(pathToMatch, { path }))[0] || {}).path;
```

Comment:
This component is almost one-to-one with the terra-application-layout's RoutingMenu. Should we match some of the implementation? Like maintaining the `getSelectableChildren` & `buildSideMenuItems` as static?
Reference: [getSelectableChildren](https://github.com/cerner/terra-framework/blob/7fb36175aa4c9ec52f1d80255041b08777ffa46c/packages/terra-application-layout/src/menu/RoutingMenu.jsx#L90) && [buildSideMenuItems](https://github.com/cerner/terra-framework/blob/7fb36175aa4c9ec52f1d80255041b08777ffa46c/packages/terra-application-layout/src/menu/RoutingMenu.jsx#L48)

Reply:
What is the benefit? That makes the implementation public on the object.. do we want that?

Reply:
Benefit that implementation would be consistent incase an enhancement was made or a bug was found in the terra-applicaiton-layout's implementation and we need to make the same changes here. However, I'm unsure about it making the implementation public. @tbiethman @dkasper-was-taken do you know?

Reply:
I'm okay with this as-is.